### DISCUSSION TITLE
[E930] - Expected function that returns an iterable of Example objects but got: []

### DISCUSSION THREAD

**Discussion by surferzskan** [explosion/spaCy#8128] [Help: Coding & Implementations] [answered]
Hello,

I'm new to Spacy (using v3.0 on Windows 10, Python=3.8), and I'm trying to implement a pipeline with a NER model for Italian language, I have my training data in the format:
`[("text", {'entities': [(start, end, ENT), (start, end, ENT), ... ]]`

I convert the data to DocBin using the following code:

```
nlp = spacy.blank("it")  # load a new spacy model
db = DocBin()  # create a DocBin object
for text, annotations in json_data:
    example = Example.from_dict(nlp.make_doc(text), annotations)
    db.add(example.reference)
db.to_disk(os.path.join(path, f"{name}.spacy"))
```

Then I create my config.cfg with this commands:
`python -m spacy init fill-config base_config.cfg config.cfg`
`python -m spacy debug config config.cfg`

Result:
============================= Config validation =============================

===================== Config validation for [initialize] =====================

====================== Config validation for [training] ======================
✔ Config is valid

When I debug the data with:
`python -m spacy debug data config.cfg`

I received the following error:

============================ Data file validation ============================
✔ Corpus is loadable
Traceback (most recent call last):
  File "C:\MyPATH\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\MyPATH\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:MyPATH\lib\site-packages\spacy\__main__.py", line 4, in <module>
    setup_cli()
  File "C:\MyPATH\lib\site-packages\spacy\cli\_util.py", line 69, in setup_cli
    command(prog_name=COMMAND)
  File "C:\MyPATH\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\MyPATH\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\MyPATH\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\MyPATH\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\MyPATH\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\MyPATH\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:MyPATH\lib\site-packages\typer\main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "C:\MyPATH\lib\site-packages\spacy\cli\debug_data.py", line 65, in debug_data_cli
    debug_data(
  File "C:\MyPATH\lib\site-packages\spacy\cli\debug_data.py", line 108, in debug_data
    nlp.initialize(lambda: train_dataset)
  File "C:\MyPATH\lib\site-packages\spacy\language.py", line 1251, in initialize
    proc.initialize(get_examples, nlp=self, **p_settings)
  File "C:\MyPATH\lib\site-packages\spacy\pipeline\tok2vec.py", line 210, in initialize
    validate_get_examples(get_examples, "Tok2Vec.initialize")
  File "spacy/training/example.pyx", line 64, in spacy.training.example.validate_get_examples
    raise TypeError(err)
TypeError: [E930] Received invalid get_examples callback in `Tok2Vec.initialize`. Expected function that returns an iterable of Example objects but got: []

I tried to look at [#7273](url) but I didn't understand very well if it's the same scenario.

Thanks

**Comment by adrianeboyd** [comment, marked as answer]
Hi, usually this error message means that somehow `train.spacy` is empty. You can load it separately and check as described here:

https://github.com/explosion/spaCy/discussions/7281#discussioncomment-430588

Nothing looks wrong in your conversion above, but possibly you haven't provided the right path for `paths.train`?

**Reply by surferzskan** [reply]
Hi adrianeboyd,
thanks for your reply.

I managed to solve the problem, it was because I had the "max_length" parameter set to 2000 in the config file.
Putting it to 0 solved the problem.
```

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null
```

Is there a place where every parameter of the config file is explained? thank you

**Reply by adrianeboyd** [reply]
Ah, thanks for replying with the solution, that's something we should also remember to bring up as a possibility when it looks like no documents are found.

The configs are flexible / modular enough that there's not one central place everything is documented, but most of the built-in registered functions aside from architectures like `spacy.Corpus.v1` are documented starting here: https://spacy.io/api/top-level#registry

The architectures are documented separately here: https://spacy.io/api/architectures

**Reply by surferzskan** [reply]
Thank you very much for your help.

**Reply by adrianeboyd** [reply]
You're welcome! `max_length = 2000` is really a bad/confusing default here, so we're planning to change it to `0` for the quickstart in the future.
